### PR TITLE
Upgrade and migrate Valibot integration to v0.31.0

### PR DESCRIPTION
**Pull request by fabian-hiller**
## 🎯 Changes

In Valibot v0.31.0 I will add a `parser` function to make it easier to use the schema library with tRPC. I have updated all Valibot related code, tests and documentation in this PR.

Our migration guide: https://valibot.dev/guides/migrate-to-v0.31.0/

> I am not sure if `next` is the right branch to merge into. Please let me know if you want me to update this PR.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Review by juliusmarminge — APPROVED** [review]
Nice! When will 0.31 be released?

**Comment by fabian-hiller** [comment]
Probably tomorrow or Wednesday.

**Comment by fabian-hiller** [comment]
v0.31.0 is available: https://valibot.dev/blog/valibot-v0.31.0-is-finally-available/

File: packages/server/package.json [hunk 1]
```diff
@@ -141,7 +141,7 @@
     "tslib": "^2.5.0",
     "tsx": "^4.0.0",
     "typescript": "^5.4.0",
-    "valibot": "^0.30.0",
+    "valibot": "0.31.0-rc.11",
```
Comment (juliusmarminge) [inline packages/server/package.json:144]: Can you use the stable version please

**Review by juliusmarminge — CHANGES_REQUESTED** [review]

File: packages/server/package.json (hunk 1 above)
Comment (fabian-hiller) [reply packages/server/package.json]: Done

File: packages/tests/server/outputParser.test.ts [hunk 2]
```diff
@@ -170,11 +170,14 @@ test('valibot transform', async () => {
...
+      .input(v.parser(v.union([v.string(), v.number()])))
       .output(
-        wrap(
+        v.parser(
           v.object({
-            input: v.transform(v.string(), (s) => s.length),
+            input: v.pipe(
+              v.string(),
+              v.transform((s) => s.length),
+            ),
```
Comment (juliusmarminge) [inline packages/tests/server/outputParser.test.ts:180]: I love this new api

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**